### DISCUSSION TITLE
Removing a SharePoint redirect

### DISCUSSION THREAD

**Discussion by OrangeNova** [pnp/powershell#4334] [Q&A] [answered]
I've been asked to remove an old SharePoint redirect; the tools MS gave were less than helpful and Google lead me here.

I've got as far as creating the app registration, and successfully connecting.  That's where I am lost.

If anyone can even point me in the correct direction, I'd be extremely grateful.

Regards,

-Martin

**Comment by Barbarur** [comment]
Would this help? https://learn.microsoft.com/en-us/sharepoint/manage-site-redirects

It shows how to delete the redirect sites.

**Reply by OrangeNova** [reply]
That's the article I started with that ended up with me landing here.

I couldn't make the initial connection to SharePoint, Google'd solutions (none of which helped), found advice to use PNP PowerShell.

Thanks for the suggestion, anyway. :)

**Reply by Barbarur** [reply]
The documentation uses a different module. https://learn.microsoft.com/en-us/powershell/sharepoint/sharepoint-online/connect-sharepoint-online 

You can install it using:

Install-Module -Name Microsoft.Online.SharePoint.PowerShell

Then connect to SharePoint using:

Connect-SPOService -Url https://contoso-admin.sharepoint.com -Credential admin@contoso.com

Finally, follow the steps of the documentation of managing redirect.


The PnP module can do the same (and much more), but for this task that you must do, I would stick with SPO module. Then later with time I would dig into PnP which will help you with more complex tasks that SPO module can't do.

**Reply by OrangeNova** [reply]
I'm sorry, I haven't coffee'd yet, I was as clear as mud.

This is where things fell apart.
Connect-SPOService -Url https://contoso-admin.sharepoint.com -Credential admin@contoso.com 

The first attempt showed I needed to use modern authentication, a tenant specific endpoint.  That for me a 400 error.  The answers I found to that were to use this module.

It will likely be tomorrow before I can try the MS module again.  I do not have time scheduled for anything SharePoint at the moment.  This was supposed to be a site for someone to prototype with, they decided they wanted the site renamed.  Now they don't like the redirect.  This is the last time I divert time to help anyone here. 😑

**Reply by Barbarur** [reply, marked as answer]
You are using MFA, then you cannot add your credentials as parameter.

To allow the sign-in window to pop and use MFA use the command as below.

Connect-SPOService -Url https://contoso-admin.sharepoint.com

**Reply by OrangeNova** [reply]
I've finally managed to get back on to this.  Don't ask me about Exchange on-prem. 🤬

Using Connect-SPOService -Url https://contoso-admin.sharepoint.com brings up a browser sign in page but still returns the 400 error.

**Reply by OrangeNova** [reply]
> You are using MFA, then you cannot add your credentials as parameter.
> 
> To allow the sign-in window to pop and use MFA use the command as below.
> 
> Connect-SPOService -Url https://contoso-admin.sharepoint.com

The only thing that was stopping this working was that I was using PowerShell 7.  It works fine with PowerShell 2.

Go figure.

Edit to add:  Thank you very much!  Your help was gratefully received.
